src/constructorof.md

```diff
@@ -32,17 +32,17 @@ julia> constructorof(S)(1,2,4)
 ERROR: AssertionError: a + b == checksum
 ```
 Instead `constructor` can be any object that satisfies the following properties:
-* It must be possible to reconstruct an object from the `NamedTuple` returned by
-`getproperties`:
+* It must be possible to reconstruct an object from the `Tuple` returned by
```

Comment:
Shouldn't this still be `getproperties`? I'm a little confused about how this is going to work. We have `getproperties` methods that let us ignore `undefined` fields and provide multiple `constructorof` implementations depending on the number of fields.

Reply:
The idea was, that `constructorof` should accept **all** private + public fields. Additional methods are optional. But maybe that is a bad idea?

Reply:
The problem is accessing undefined fields is an error, and they exist in  few Base objects. But we can special case `getfields` instead of `getoroperties` to check if fields are defined?

Reply:
You can not have undefined fields in user defined types right? So we could take care of all builtin types that have these?

Reply:
Yes, I think we could special case in `getfields` for undefined fields for Base objects.

Reply:
There are some objects like that here that currently use `getproperties`